core/arch/arm/mm/tee_mmu.c

```diff
@@ -227,27 +230,42 @@ static TEE_Result tee_mmu_umap_set_vas(struct tee_mmu_info *mmu)
 
 static unsigned int asid_alloc(void)
 {
+	uint32_t exceptions = cpu_spin_lock_xsave(&g_asid_spinlock);
+	unsigned int r = 0;
 	int i;
 
 	bit_ffc(g_asid, MMU_NUM_ASIDS, &i);
-	if (i == -1)
-		return 0;
-	bit_set(g_asid, i);
+	if (i == -1) {
+		r = 0;
+	} else {
+		bit_set(g_asid, i);
+
+		r = (i + 1) * 2;
+	}
 
-	return (i + 1) * 2;
+	cpu_spin_unlock_xrestore(&g_asid_spinlock, exceptions);
+	return r;
 }
 
 static void asid_free(unsigned int asid)
 {
+	uint32_t exceptions = cpu_spin_lock_xsave(&g_asid_spinlock);
+
 	/* Only even ASIDs are supposed to be allocated */
 	assert(!(asid & 1));
 
 	if (asid) {
 		int i = (asid - 1) / 2;
 
+		if (!(i < MMU_NUM_ASIDS && bit_test(g_asid, i))) {
```

Comment:
minor: more readable as `(i >= MMU_NUM_ASIDS || !bit_test(g_asid, i))`

Reply:
Sorry, this should be removed, it was just debug code.